pb/metainfo.proto

```diff
@@ -781,10 +790,51 @@ message ObjectFinishMoveRequest {
     repeated EncryptedKeyAndNonce new_segment_keys = 5;
 }
 
+message ObjectFinishMoveResponse { }
+
+// Server side copy.
+// Has the same fields as move.
+message ObjectBeginCopyRequest {
+    RequestHeader header = 15;
+
+    bytes bucket = 1;
+    bytes encrypted_object_key = 2;
+
+    bytes new_bucket = 3;
+    bytes new_encrypted_object_key = 4;
+}
+
+// The satellite returns metadata of the source object,
+// so that the client can re-encrypt it.
+message ObjectBeginCopyResponse {
+    bytes source_stream_id = 1 [(gogoproto.customtype) = "StreamID", (gogoproto.nullable) = false];
+
+    bytes encrypted_metadata_key_nonce = 2 [(gogoproto.customtype) = "Nonce", (gogoproto.nullable) = false];
+    bytes encrypted_metadata_key = 3;
+    repeated EncryptedKeyAndNonce segment_keys = 4;
+    encryption.EncryptionParameters encryption_parameters = 5;
+}
+
+// Uplink uploads the newly encrypted metadata fore the destion object,
+// so that the satellite can finish the copy.
+message ObjectFinishCopyRequest {
+    RequestHeader header = 15;
+
+    // Stream ID of source object
+    bytes stream_id = 1 [(gogoproto.customtype) = "StreamID", (gogoproto.nullable) = false];
+
+    // Newly encrypted data of the copy destiation
+    bytes new_bucket = 6;
```

Comment:
can make numbers orderd? it will be a bit cleaner. I'm not sure why we have this strange ordering in move request